src/datasources/jwt/jwt.module.ts

```diff
@@ -0,0 +1,43 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { JwtService } from '@/datasources/jwt/jwt.service';
+import { IJwtService } from '@/domain/interfaces/jwt-api.interface';
+import { Global, Module } from '@nestjs/common';
+import * as jwt from 'jsonwebtoken';
+
+// Use inferred type
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function jwtClientFactory(configurationService: IConfigurationService) {
+  const secret = configurationService.getOrThrow<string>('jwt.secret');
+
+  return {
+    sign: (
+      payload: string | Buffer | object,
+      options?: jwt.SignOptions,
+    ): string => {
+      return jwt.sign(payload, secret, options);
+    },
+    verify: <T>(
+      token: string,
+      options?: Omit<jwt.VerifyOptions, 'complete'>,
+    ): T => {
+      // As we are not `complete` decoding, it will be T if verified
+      return jwt.verify(token, secret, options) as T;
```

Comment:
Should we opt for JWTs, we shoul also set [the issuer, etc.](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1) to confirm the expected entity signed it.